generator/ServiceClientGeneratorLib/Generators/Marshallers/AWSQueryEC2ResponseUnmarshaller.tt

```diff
@@ -72,6 +72,8 @@ namespace <#=this.Config.Namespace #>.Model.Internal.MarshallTransformations
 					if (member.IsMap || member.IsList)
 					{
 #>
+                        context.KeyXmlName = "<#=member.Shape.KeyMarshallName#>";
```

Comment:
This mechanism of storing the current key and value name for the shape being marshalled to the context seems like it would be broken if we had values that were maps as well that had different `KeyMarshallName`. When we made the recursive call to to the child map being marshalled the context would be changed and then when we process the next element in the parent map we would have different `KeyXmlName` or `ValueXmlName`.

Reply:
Going to have to update this, to handle nested xmlNames on key/values. Will push an update with that fix later